tuned/profiles/functions/function_get_queue_count.py

```diff
@@ -0,0 +1,23 @@
+import tuned.logs
+from . import base
+from tuned.utils.commands import commands
+
+log = tuned.logs.get()
+
+class get_queue_count(base.Function):
+	"""
+	Checks whether the user has specified a queue count for net devices. If
+        not, return the number of housekeeping CPUs.
+	"""
+	def __init__(self):
+		# 1 argument
+		super(get_queue_count, self).__init__("get_queue_count", 1, 1)
+
+	def execute(self, args):
+		if not super(get_queue_count, self).execute(args):
+			return None
+		if args[0].isdigit():
+			return args[0]
+		(ret, out) = self._cmd.execute(["nproc"])
```

Comment:
I would use the  /sys/devices/system/cpu/online file, nproc may be bound by the current process limits. nproc --all can be used, but better to use the sysfs, I think.
tuned already has some cpulist_online class, maybe will help.

Reply:
"/sys/devices/system/cpu/online" includes all the online CPUs (isolated+HK), what we want is to get only the number of HK CPUs.